extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriLightHandler.java

```diff
@@ -0,0 +1,322 @@
+        JsonObject root;
+        JsonArray array;
+        JsonObject attributes;
+
+        public LightData() {
+            root = new JsonObject();
+            array = new JsonArray();
+            attributes = new JsonObject();
+            array.add(attributes);
+            root.add(LIGHT, array);
+        }
+
+        public LightData(JsonElement json) {
+            try {
+                root = json.getAsJsonObject();
+                array = root.getAsJsonArray(LIGHT);
+                attributes = array.get(0).getAsJsonObject();
+            } catch (JsonSyntaxException e) {
+                logger.error("JSON error: {}", e.getMessage(), e);
+            }
+        }
+
+        public LightData setBrightness(PercentType brightness) {
+            attributes.add(DIMMER, new JsonPrimitive(Math.round(brightness.floatValue() / 100.0f * 254)));
+            return this;
+        }
+
+        public PercentType getBrightness() {
+            int b = attributes.get(DIMMER).getAsInt();
+            if (b == 1) {
+                return new PercentType(1);
+            }
+            return new PercentType((int) Math.round(b / 2.54));
+        }
+
+        public boolean getReachabilityStatus() {
+            if (attributes.get(REACHABILITY_STATE) != null) {
+                return attributes.get(REACHABILITY_STATE).getAsInt() == 1;
+            } else {
+                return false;
+            }
+        }
+
+        public LightData setTransitionTime(int seconds) {
+            attributes.add(TRANSITION_TIME, new JsonPrimitive(seconds));
+            return this;
+        }
+
+        @SuppressWarnings("unused")
+        public int getTransitionTime() {
+            return attributes.get(TRANSITION_TIME).getAsInt();
+        }
+
+        private final static double[] X = new double[] { 33137.0, 30138.0, 24933.0 };
+        private final static double[] Y = new double[] { 27211.0, 26909.0, 24691.0 };
+
+        public LightData setColorTemperature(PercentType c) {
+            double percent = c.doubleValue();
+
+            long x, y;
+            if (percent < 50.0) {
+                double p = percent / 50.0;
+                x = Math.round(X[0] + p * (X[1] - X[0]));
+                y = Math.round(Y[0] + p * (Y[1] - Y[0]));
+            } else {
+                double p = (percent - 50) / 50.0;
+                x = Math.round(X[1] + p * (X[2] - X[1]));
+                y = Math.round(Y[1] + p * (Y[2] - Y[1]));
+            }
+            logger.debug("New color temperature: {},{} ({} %)", x, y, percent);
+
+            attributes.add(COLOR_X, new JsonPrimitive(x));
+            attributes.add(COLOR_Y, new JsonPrimitive(y));
+            return this;
+        }
+
+        PercentType getColorTemperature() {
+            JsonElement colorX = attributes.get(COLOR_X);
+            if (colorX != null) {
+                double x = attributes.get(COLOR_X).getAsInt();
+                double value = 0.0;
+                if (x > X[1]) {
```

Comment:
I didn't do any color temperature calculation before but it surprises me that set takes the `x` & `y` into account but the get only the `x`. Can you validate?